genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaJobSearchServiceImpl.java

```diff
@@ -362,4 +363,43 @@ public String getJobHost(@NotBlank final String jobId) throws GenieException {
             throw new GenieNotFoundException("No job execution found for id " + jobId);
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Long getActiveJobCountByUserName(@NotBlank final String userName) {
```

Comment:
There are much simpler ways to do this with Spring Data JPA. The reason that in the job search we use this dynamic build up via JPA criteria builder is because there are some parameters that may or may not be there hence we use if statements to do it.

In this instance you know exactly what parameters you want to pass in so you can leverage their nice query builder methods that do it all for you.

See https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods for more.